### PR TITLE
update large files to use jest

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/large_files/test/pretest.ts [hunk 1]
```diff
@@ -1,36 +1,11 @@
...
-    await generateTestFileOfSize(150, 'MiB', 'manual');
-
     execSync(`dfx canister uninstall-code backend || true`, {
         stdio: 'inherit'
     });
 
-    execSync(`dfx deploy`, {
-        stdio: 'inherit'
-    });
+    execSync(`dfx canister create backend`);
```
Comment (bdemann) [inline examples/large_files/test/pretest.ts:8]: Add a comment explaining why we are only creating the canister here

File: examples/large_files/test/tests.ts [hunk 2]
```diff
@@ -1,233 +1,90 @@
...
-        {
-            name: 'redeploy',
-            prep: async () => {
-                await generateTestFileOfSize(1, 'KiB');
-                await generateTestFileOfSize(10, 'KiB');
-                await generateTestFileOfSize(100, 'KiB');
-                execSync(`dfx deploy --upgrade-unchanged`, {
+    return () => {
+        beforeAll(async () => {
+            // Ensure all files from previous runs are cleared out
```
Comment (bdemann) [inline examples/large_files/test/tests.ts:23]: My thoughts are that it would be nice to clear out the files after the test so they aren't clutter, but we should probably remove them before so that we are confident they aren't there (if the process was aborted before clean up) but it looks a tad redundant. Thoughts?

File: examples/large_files/test/tests.ts [hunk 3]
```diff
@@ -1,233 +1,90 @@
...
-        generateTest('manual test', origin, 'test150MiB', 'manual'),
-        // TODO CI CD isn't working with the 2GiB tests so we're just going to have this one for local tests.
-        {
-            name: 'deploy',
-            prep: async () => {
-                await rm(join('assets', 'auto'), {
-                    recursive: true,
-                    force: true
+        describe(
+            'verify files specified in dfx.json exists after initial deploy',
```
Comment (bdemann) [inline examples/large_files/test/tests.ts:48]: exist

File: examples/large_files/test/tests.ts [hunk 4]
```diff
@@ -1,233 +1,90 @@
...
-        ...generateStandardFileTests('Upload', origin),
-        {
-            name: 'redeploy',
-            prep: async () => {
-                await generateTestFileOfSize(1, 'KiB');
-                await generateTestFileOfSize(10, 'KiB');
-                await generateTestFileOfSize(100, 'KiB');
-                execSync(`dfx deploy --upgrade-unchanged`, {
+    return () => {
+        beforeAll(async () => {
```
Comment (lastmjs) [inline examples/large_files/test/tests.ts:22]: The manual folder needs to be cleared out as well

File: examples/large_files/test/auto_tests.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+const autoGenAutoUploadTests: [number, Unit][] = [
+    // Edge Cases
+    [0, 'B'],
+    [1, 'B'],
+    [120 * 1024 * 1024 + 1, 'B'], // One more byte than can be processed in a single hash_file_by_parts call
+    [2_000_000 + 1, 'B'], // One more byte that the message chunk size
+
+    // General Cases
+    // TODO Add tests for huge files after https://github.com/wasm-forge/stable-fs/issues/2 is resolved
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:16]: Remove TODO

File: examples/large_files/test/auto_tests.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    [10, 'KiB'],
+    [100, 'KiB'],
+    [1, 'MiB'],
+    [10, 'MiB'],
+    [100, 'MiB'],
+    [250, 'MiB'],
+    [1, 'GiB']
+];
+
+const permanentFilesTests: [string][] = [
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:27]: Change to just an array of strings

File: examples/large_files/test/auto_tests.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
+import { describe } from '@jest/globals';
+import { please, Test } from 'azle/test/jest';
+
+import { Unit } from '../../../scripts/file_generator';
+import { generateTestFileOfSize } from './generate_test_files';
+import { getAutoGeneratedFileName, verifyUpload } from './tests';
+
+const autoGenAutoUploadTests: [number, Unit][] = [
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:8]: Rename

File: examples/large_files/test/auto_tests.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    ['photos/things/utah-teapot.jpg'],
+    ['text/subfolder/deep-sub-folder/deep.txt'],
+    ['text/subfolder/sibling-deep-sub-folder/deep.txt'],
+    ['text/subfolder/other-thing.txt'],
+    ['text/thing.txt'],
+    ['text/thing.txt']
+];
+
+const fileRenameDuringUploadTests: [string, string][] =
+    // TODO this is a bit of a pickle one
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:42]: Remove

File: examples/large_files/test/auto_tests.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * @param origin - The origin URL or path used in the tests.
+ * @returns A suite of Jest tests.
+ */
+export function getDfxConfigFileTests(origin: string): Test {
+    return () => {
+        describe.each(permanentFilesTests)(
+            'permanent files with same names locally as on canister',
+            (canisterPath) => {
+                verifyUpload(origin, canisterPath, 'permanent');
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:75]: Reorder to be src dest

File: examples/large_files/test/auto_tests.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+            }
+        );
+
+        describe.each(fileRenameDuringUploadTests)(
+            'permanent files with different names locally than on canister',
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:80]: Add "uploaded as canister was deployed"

File: examples/large_files/test/auto_tests.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+ * automatically uploaded to the canister as specified in the `dfx.json`
+ * configuration.
+ *
+ * @param origin - The origin URL or path used in the tests.
+ * @returns A suite of Jest tests.
+ */
+export function getDfxConfigFileTests(origin: string): Test {
+    return () => {
+        describe.each(permanentFilesTests)(
+            'permanent files with same names locally as on canister',
```
Comment (lastmjs) [inline examples/large_files/test/auto_tests.ts:73]: Add "uploaded as canister was deployed"

File: examples/large_files/test/huge_file_tests.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+            'remove all other auto generated files so there is room for huge files',
+            async () => {
+                await rm(join('assets', 'auto'), {
+                    recursive: true,
+                    force: true
+                });
+            }
+        );
+
+        describe.each(hugeAutoGenAutoUploadTests)(
```
Comment (lastmjs) [inline examples/large_files/test/huge_file_tests.ts:25]: Rename to `hugeAutoGenAutoUploadTests` to something not ending in `Tests`

File: examples/large_files/test/huge_file_tests.ts [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+                    stdio: 'inherit'
+                });
+            }
+        );
+
+        describe.each(hugeAutoGenAutoUploadTests)(
+            'verify huge files were uploaded correctly',
+            (size, units) => {
+                const fileName = getAutoGeneratedFileName(size, units);
+                verifyUpload(origin, fileName, 'auto', undefined);
```
Comment (lastmjs) [inline examples/large_files/test/huge_file_tests.ts:52]: Get rid of the undefined argument

File: examples/large_files/test/manual_tests.ts [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+import { expect, it, please, Test } from 'azle/test/jest';
+import { execSync } from 'child_process';
+import { join } from 'path';
+
+import { Unit } from '../../../scripts/file_generator';
+import { generateTestFileOfSize } from './generate_test_files';
+import { getAutoGeneratedFileName, verifyUpload } from './tests';
+
+export function manualTests(origin: string): Test {
+    const autoGenManualUploadTests: [number, Unit][] = [[150, 'MiB']];
```
Comment (lastmjs) [inline examples/large_files/test/manual_tests.ts:11]: Rename similar to other renames, not ending in Tests, they are something besides tests.

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/large_files/test/tests.ts (hunk 2 above)
Comment (bdemann) [reply examples/large_files/test/tests.ts:23]: We decided that both is find

File: examples/large_files/test/tests.ts (hunk 4 above)
Comment (bdemann) [reply examples/large_files/test/tests.ts:22]: done

File: examples/large_files/test/auto_tests.ts (hunk 5 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts:16]: done

File: examples/large_files/test/auto_tests.ts (hunk 6 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts:27]: done

File: examples/large_files/test/auto_tests.ts (hunk 7 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts:8]: done

File: examples/large_files/test/auto_tests.ts (hunk 8 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts:42]: done

File: examples/large_files/test/huge_file_tests.ts (hunk 12 above)
Comment (bdemann) [reply examples/large_files/test/huge_file_tests.ts:25]: done

File: examples/large_files/test/huge_file_tests.ts (hunk 13 above)
Comment (bdemann) [reply examples/large_files/test/huge_file_tests.ts:52]: done

File: examples/large_files/test/manual_tests.ts (hunk 14 above)
Comment (bdemann) [reply examples/large_files/test/manual_tests.ts:11]: done

File: examples/large_files/test/auto_tests.ts (hunk 10 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts:80]: done

File: examples/large_files/test/auto_tests.ts (hunk 11 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts:73]: done

File: examples/large_files/test/auto_tests.ts (hunk 7 above)
Comment (lastmjs) [reply examples/large_files/test/auto_tests.ts]: Should still be plural

File: examples/large_files/test/huge_file_tests.ts (hunk 12 above)
Comment (lastmjs) [reply examples/large_files/test/huge_file_tests.ts]: Plural

File: examples/large_files/test/manual_tests.ts (hunk 14 above)
Comment (lastmjs) [reply examples/large_files/test/manual_tests.ts]: Plural

**Review by lastmjs — DISMISSED** [review]

File: examples/large_files/test/manual_tests.ts (hunk 14 above)
Comment (bdemann) [reply examples/large_files/test/manual_tests.ts]: done

File: examples/large_files/test/huge_file_tests.ts (hunk 12 above)
Comment (bdemann) [reply examples/large_files/test/huge_file_tests.ts]: done

File: examples/large_files/test/auto_tests.ts (hunk 7 above)
Comment (bdemann) [reply examples/large_files/test/auto_tests.ts]: done
